src/exit_codes.rs

```diff
@@ -17,7 +19,11 @@ impl From<ExitCode> for i32 {
 
 impl ExitCode {
     fn is_error(self) -> bool {
-        self != ExitCode::Success
+        match self {
+            ExitCode::Success => false,
+            ExitCode::HasMatch(has_match) => !has_match,
+            _ => true,
+        }
```

Comment:
Hm. Maybe it makes sense to implement `is_error` in terms of `from(code: ExitCode)`? This way, we don't have to duplicate the logic. And this is not a function that is performance-critical.

Reply:
Good catch; definitely sounds better than my solution!